libbeat/esleg/eslegclient/connection.go

```diff
@@ -129,28 +129,26 @@ func NewConnection(s ConnectionSettings) (*Connection, error) {
 		}
 	}
 
-	var httpClient esHTTPClient
 	// when dropping the legacy client in favour of the official Go client, it should be instrumented
 	// eg, like in https://github.com/elastic/apm-server/blob/7.7/elasticsearch/client.go
+	transp := apmelasticsearch.WrapRoundTripper(&http.Transport{
+		Dial:            dialer.Dial,
+		DialTLS:         tlsDialer.Dial,
+		TLSClientConfig: s.TLS.ToConfig(),
+		Proxy:           proxy,
+		IdleConnTimeout: s.IdleConnTimeout,
+	})
+
+	var httpClient esHTTPClient
 	httpClient = &http.Client{
-		Transport: apmelasticsearch.WrapRoundTripper(&http.Transport{
-			Dial:            dialer.Dial,
-			DialTLS:         tlsDialer.Dial,
-			TLSClientConfig: s.TLS.ToConfig(),
-			Proxy:           proxy,
-			IdleConnTimeout: s.IdleConnTimeout,
-		}),
-		Timeout: s.Timeout,
+		Transport: transp,
+		Timeout:   s.Timeout,
 	}
 
 	if s.Kerberos.IsEnabled() {
 		c := &http.Client{
```

Comment:
I think we can now remove this statement and use `httpClient` instead of `c` as the 2nd argument to `kerberos.NewClient` below now?

Reply:
No, we cannot. `kerberos.NewClient` expects an `http.Client`, but the client above is an `esHTTPClient`.